lib/foreman_maintain/concerns/firewall/maintenance_mode.rb

```diff
@@ -0,0 +1,38 @@
+module ForemanMaintain
+  module Concerns
+    module Firewall
+      module MaintenanceMode
+        def notify_and_ask_to_install_firewall_utility
+          puts 'Unable to find nftables or iptables!'
+          question, pkg = question_and_pkg_name
+          answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
+          if answer == :yes
+            packages_action(:install, pkg)
+            feature(:instance).firewall.enable_maintenance_mode
+          end
+        end
+
+        def can_install_nft?
+          # The nftables is default from EL8 and Debian 10(Buster)
+          (el? && el_major_version >= 8) ||
+            (debian? && deb_major_version >= 10) ||
+            (ubuntu? && ubuntu_major_version.to_i >= 22)
+        end
+
+        def question_and_pkg_name
+          question = 'Do you want to install missing netfilter utility '
+          pkg_to_install = []
+          if can_install_nft?
+            question << 'nftables?'
+            pkg_to_install << 'nftables'
+          else
+            question << 'iptables?'
+            pkg_to_install << 'iptables'
+          end
+          question << "\nand start maintenance mode?"
+          [question, pkg_to_install]
```

Comment:
You are modifying the string in place here (using <<) which conflicts with frozen strings (default starting with Ruby 3.0). I'd avoid it altogether and reduce it to:
```suggestion          
          pkg_to_install = can_install_nft? ? 'nftables' : 'iptables'
          question = "Do you want to install missing netfilter utility #{pkg_to_install}?\nand start maintenance mode?"
          [question, [pkg_to_install]]
```